File: Go/RecursiveFibonacciService.go
package main

import (
	"Fibonacci/common"
)

type RecursiveFibonacciService struct {
}

func (s RecursiveFibonacciService) GetNumbers(count int) []int {
	return getNumbersRecursive(nil, count)
}

func getNumbersRecursive(numbers []int, count int) []int {
	var outNumbers []int

	if len(numbers) > 0 {
		var x, y int = common.GetPriorValues(numbers)
		var nextValue int = common.GetNextValue(x, y)
		outNumbers = append(numbers, nextValue)
	} else {
		outNumbers = append(numbers, 0)
	}

	if len(outNumbers) < count {
		outNumbers = getNumbersRecursive(outNumbers, count)
	}

	return outNumbers
}

File: CSharp/Fibonacci.Lib/RecursiveFibonacciService.cs
namespace Fibonacci.Lib
{
    public class RecursiveFibonacciService : IFibonacciService
    {
        public IEnumerable<int> GetNumbers(int count) => GetNumbersRecursive(count, new List<int>());

        private List<int> GetNumbersRecursive(int count, IEnumerable<int> numbers)
        {
            var outNumbers = numbers.ToList();
            if (numbers.Any())
            {
                var priorValues = FibonacciHelper.GetPriorValues(numbers);
                var nextValue = FibonacciHelper.GetNextNumber(priorValues.Item1, priorValues.Item2);
                outNumbers.Add(nextValue);
            }
            else
            {
                outNumbers.Add(0);
            }

            if (outNumbers.Count < count)
            {
                outNumbers = GetNumbersRecursive(count, outNumbers);
            }

            return outNumbers;
        }
    }
}
